migrations/blocked_tasks.js

```diff
@@ -0,0 +1,83 @@
+// constants
+finishedStatuses = ["success","failed"]
+
+function taskIsBlocked(task) {
+    for(dep of task.depends_on) {
+        if (dep.unattainable) {
+            return true
+        }
+    }
+
+    return false
+}
+
+function unattainableResolved(task) {
+    for (dep of task.depends_on) {
+        if (dep.unattainable == null) {
+            return false
+        }
+    }
+    return true
+}
+
+//
+// example invocation:
+//
+// startingDate = ISODate("2016-01-01T00:00:00Z")
+// millisecondsPerDay = 86400000
+// delta = millisecondsPerDay * 1
+// sleepMilliseconds = 100
+// migrate(startingDate, delta, sleepMilliseconds)
+
+function migrate(startingDate, delta, sleepMilliseconds) {
+    while(startingDate < new Date()) {
+        loops = 0
+        tasksSize = 0
+        while (true) {
+            tasks = db.tasks.find({"create_time": {"$gte": startingDate, "$lte": new Date(startingDate.getTime() + delta)}, "status":"undispatched", "depends_on.status": {"$in": ["success", "failed", "", "*"]}, "depends_on": {"$elemMatch":{"unattainable": {"$exists": false}}}}, {"depends_on":1}).toArray()
+            printjson(loops++)
+            if (tasks.length == 0 || tasks.length == tasksSize) {
+                break
+            }
+            tasksSize = tasks.length
+
+            for (i=0; i < tasks.length; i++) {
+                taskUpdated = false
+                dependsOn = tasks[i].depends_on
+                for (j=0; j < dependsOn.length; j++) {
+                    if(dependsOn[j].unattainable != null) {
+                        continue
+                    }
+                    
+                    depTask = db.tasks.findOne({"_id": dependsOn[j]._id}, {"status": 1, "depends_on": 1});
```

Comment:
would it be possible to collect all of the _id fields we need and do a $in, so we only have to do one query?

Reply:
done.